app/presenters/wpcc/earnings_description.rb

```diff
@@ -0,0 +1,21 @@
+module Wpcc
+  module EarningsDescription
+    def formatted_eligible_salary
+      formatted_currency(object.eligible_salary, precision: 0)
+    end
+
+    def earnings_description
+      contribution_preference = session[:contribution_preference]
+      description_key = earnings_description_key
+
+      t(
+        "wpcc.#{description_key}.description_#{contribution_preference}",
+        eligible_salary: formatted_eligible_salary
+      )
+    end
+
+    def earnings_description_key
+      raise NotImplementedError
```

Comment:
Is there scenario/spec which will cause this `NotImplementedError` to be raised? Should we add a spec to cover this?

Reply:
No, we shouldn't add a test for this because this whole module is an interface to be included in two presenter classes. On the classes which include this module, this method is tested. And this module has a shared example which test this interface.

Reply:
@tomas-stefano oh okay thanks for explanation.